app/models/service_alerts/alert.rb

```diff
@@ -2,27 +2,46 @@ module ServiceAlerts
   class Alert < ActiveRecord::Base
     include DatedFeed
     include HtmlSanitizer
+    include OraclePrimaryHelper
 
-    self.table_name = 'service_alerts'
+    self.table_name = 'PS_UC_CLC_SRVALERT'
+    self.primary_key = 'uc_clc_id'
 
-    attr_accessible :title, :snippet, :body, :publication_date, :display, :splash
+    def self.attributeDefaults
+      {uc_alrt_title:' ', uc_alrt_snippt:' ', uc_alrt_body:' ', uc_alrt_display:false, uc_alrt_splash:false, uc_alrt_pubdt:Time.zone.today.in_time_zone.to_datetime}
+    end
+
+    attr_accessible :uc_clc_id, :uc_alrt_title, :uc_alrt_snippt, :uc_alrt_pubdt, :uc_alrt_display
+    attr_accessible :uc_alrt_splash, :uc_alrt_body, :created_at, :updated_at
+    attr_accessible :title, :display, :body, :publication_date, :snippet
+
+    alias_attribute :title, :uc_alrt_title
+    alias_attribute :display, :uc_alrt_display
+    alias_attribute :body, :uc_alrt_body
+    alias_attribute :publication_date, :uc_alrt_pubdt
+    alias_attribute :snippet, :uc_alrt_snippt
 
-    validates :title, presence: true
-    validates :body, presence: true
-    validates :publication_date, presence: true
+    validates :uc_alrt_title, presence: true
+    validates :uc_alrt_body, presence: true
 
-    after_initialize :set_default_publication_date
+    before_save :set_default_values
+    before_create :set_id
+
+    if ENV["RAILS_ENV"]=='production' or (ENV["RAILS_ENV"]=='development' and Settings.devdb.adapter == 'oracle_enhanced')
+      set_boolean_columns :uc_alrt_display, :uc_alrt_splash
+    end
```

Comment:
@frankiepun - It looks like we're using this logic in many places. Perhaps we should create a function that can encapsulate this, and re-use it.

I recommend defining the method in OraclePrimaryHelper as follows.
```
def primary_database_is_oracle
  ENV["RAILS_ENV"]=='production' || (ENV["RAILS_ENV"]=='development' && Settings.devdb.adapter == 'oracle_enhanced')
end
```

Also you'll notice I'm using `&&` instead of `and`. See [StackOverflow - Difference Between AND and && in Ruby](https://stackoverflow.com/questions/1426826/difference-between-and-and-in-ruby)

In fact, I'm curious why we're not using the `Rails.env` support for this. The 'RAILS_ENV' environment variable might not always be present.

```
def primary_database_is_oracle
  Rails.env.production? || (Rails.env.development? && Settings.devdb.adapter == 'oracle_enhanced')
end
```

Reply:
We changed it avoid checking environment variables. It just checked if the ActiveRecord's connection is OracleEnhancedAdapter.